src/client/src/Client.c

```diff
@@ -862,6 +862,52 @@ STATUS putKinesisVideoFragmentMetadata(STREAM_HANDLE streamHandle, PCHAR name, P
     return retStatus;
 }
 
+
+/**
+ * Puts an event & associated custom data into the Kinesis Video stream.
+ */
+
+STATUS putKinesisVideoEventMetadata(STREAM_HANDLE streamHandle, STREAM_EVENT_TYPE event, PStreamEventMetadata pStreamEventMetadata)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    BOOL releaseClientSemaphore = FALSE, releaseStreamSemaphore = FALSE;
+    PKinesisVideoStream pKinesisVideoStream = FROM_STREAM_HANDLE(streamHandle);
+
+    DLOGS("Put event into an Kinesis Video stream.");
+
+    CHK(pKinesisVideoStream != NULL && pKinesisVideoStream->pKinesisVideoClient != NULL, STATUS_NULL_ARG);
+    if(pStreamEventMetadata != NULL) {

```

Comment:
The event meta data is optional. It is sanitized here because this API has an expectation that the meta data will only have complete pairs, and a limited number of them.

Reply:
The event meta data is optional. It is sanitized here because this API has an expectation that the meta data will only have complete pairs, and a limited number of them.

Reply:
Most of the code assumes the semantical validation is done in the internal API. Not a biggie but it's a cleaner separation of context. Nothing prevents checking the contents in the internal API which might or might not be called from elsewhere - just a more robust approach for change-proofing in mu opinion.